contracts/staking/DonationsStaking.sol

```diff
@@ -154,4 +156,8 @@ contract DonationsStaking is DAOUpgradeableContract {
 		stakingToken.approve(address(stakingContract), type(uint256).max); //we trust the staking contract
 		stakingToken.approve(address(uniswap), type(uint256).max); // we trust uniswap router
 	}
+
+	function getRouter() public view override returns (Uniswap) {
+		return Uniswap(nameService.getAddress("UNISWAP_ROUTER"));
+	}
```

Comment:
this is not correct
the methods the uniswaphelper expects the "self" contract to have are
nameService()
maxLiquidityPercentageSwap()